### PR TITLE
Documentation for ramble mirror

### PR DESCRIPTION
**Pull request by dodecatheon**
Basic documentation for mirror usage.

### CODE REVIEW

**Comment by dodecatheon** [comment]
Added tutorial, rebased, squashed changes. Ready for review.

File: lib/ramble/docs/mirror-config.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+To speed up installations, Spack provides the ``mirror`` command, documented
+[here](https://spack.readthedocs.io/en/latest/mirrors.html).
+
+Ramble provides a similar ``mirror`` command, in order to find workload input files
+locally first instead of in their default locations.
+
+The ``ramble mirror`` set of commands largely follow the same form as ``spack mirror`` commands
+(``add``, ``list``, ``set-url``, ``remove``, ``destroy``), with the exception of mirror creation.
+
+Since Ramble input mirrors are intended to for use by applications, and applications are used
```
Comment (douglasjacobsen) [inline lib/ramble/docs/mirror-config.rst:29]: `to for use` --> `to be used`

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]
I'd like @rfbgo to review this too, to make sure the rst format is correct for sphinx.

**Comment by rfbgo** [comment]
> I'd like @rfbgo to review this too, to make sure the rst format is correct for sphinx.

I'd rather not be the human rst format parser. Can we just have someone run the document generation and confirm it renders properly + no new warnings are added?

I am certainly able to do that, but in general would prefer it's on the contributor to uphold high standards (when they're internal)

File: lib/ramble/docs/mirror-config.rst (hunk 1 above)
Comment (dodecatheon) [reply lib/ramble/docs/mirror-config.rst]: Done

File: lib/ramble/docs/tutorials/mirrors.rst [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+
+-------------
+Mirror creation:
+-------------
+To start off with, let's set up a workspace with a basic experiment in it.
+The experiment doesn't have to be very sophisticated, but it does need to
+specify the application you want to mirror, and any application workloads
+(with sha256 sums) that you want to have mirrored.
+
+For the purpose of this demonstration, I will assume that you're using
```
[the remark this answers is not in the document]
Comment (dodecatheon) [reply lib/ramble/docs/tutorials/mirrors.rst]: This plus related wording changes, done

File: lib/ramble/docs/tutorials/mirrors.rst [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+        └── zstd
+            └── zstd-1.5.5.tar.gz -> ../_source-cache/archive/98/98e9c3d949d1b924e28e01eccb7deed865eefebf25c2f21c702e5cd5b63b85e1.tar.gz
+    
+    133 directories, 204 files
+
+
+-------------
+Mirror use:
+-------------
+Once a mirror has been created locally, follow the directions in :ref:`Mirror use <mirror-use>`_ to use
```
[the remark this answers is not in the document]
Comment (dodecatheon) [reply lib/ramble/docs/tutorials/mirrors.rst:512]: This plus related wording changes, done

File: lib/ramble/docs/mirror-config.rst [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+via Ramble workspaces, Ramble input mirrors are created via the ``ramble workspace mirror``
+command, documented below. This command creates both the ramble input file mirror and the
+spack source archive mirror, both of which are automatically restricted to mirror only those
+input files and packages required for the ramble applications used in the workspace.
+
+Once ramble input and spack software mirrors have been created, the user wishing to speed
```
[the remark this answers is not in the document]
Comment (dodecatheon) [reply lib/ramble/docs/mirror-config.rst:37]: I've reworded slightly based on your comment. Hopefully it now conveys the idea better.

**Comment by dodecatheon** [comment]
Found a broken ref after running sphinx and fixed it.

File: lib/ramble/docs/mirror-config.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (douglasjacobsen) [reply lib/ramble/docs/mirror-config.rst:37]: I'd still rather not point this at speeding up an installation. The goal of the mirror is not to simply speed up access to source downloads and input files. It also allows you to create backups if they don't exist externally, or provide access to machines without internet access.

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]
Thanks for the changes. They look better. One more nit, but otherwise LGTM.

File: lib/ramble/docs/mirror-config.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (dodecatheon) [reply lib/ramble/docs/mirror-config.rst]: More rewording.

**Comment by dodecatheon** [comment]
Rewording should have addressed requested changes.

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**